servers/dsub/jobs/controllers/utils/generator.py

```diff
@@ -0,0 +1,164 @@
+import datetime
+
+from flask import current_app, request
+from dsub.commands import ddel, dstat
+from dateutil.tz import tzlocal
+
+from jobs.common import execute_redirect_stdout
+from jobs.controllers.utils import extensions, failures, job_ids, job_statuses, labels, providers, query_parameters
+from jobs.models.query_jobs_result import QueryJobsResult
+
+_MAX_PENDING_TIME_DAYS = 7
+
+
+def generate_jobs(provider, query, create_time_max=None, offset_id=None):
+    """Get the generator of jobs according to the specified filters.
+
+        Args:
+            provider (string): type of provider
```

Comment:
string->str. The "type" in python is "str".

Reply:
Fixed.